machineconfiguration/v1/types.go

```diff
@@ -0,0 +1,389 @@
+}
+
+// MachineConfigPoolStatus is the status for MachineConfigPool resource.
+type MachineConfigPoolStatus struct {
+	// The generation observed by the controller.
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	// The current MachineConfig object for the machine config pool.
+	Configuration MachineConfigPoolStatusConfiguration `json:"configuration"`
+
+	// Total number of machines in the machine config pool.
+	MachineCount int32 `json:"machineCount"`
+
+	// Total number of machines targeted by the pool that have the CurrentMachineConfig as their config.
+	UpdatedMachineCount int32 `json:"updatedMachineCount"`
+
+	// Total number of ready machines targeted by the pool.
+	ReadyMachineCount int32 `json:"readyMachineCount"`
+
+	// Total number of unavailable (non-ready) machines targeted by the pool.
+	// A node is marked unavailable if it is in updating state or NodeReady condition is false.
+	UnavailableMachineCount int32 `json:"unavailableMachineCount"`
+
+	// Total number of machines marked degraded (or unreconcilable).
+	// A node is marked degraded if applying a configuration failed..
+	DegradedMachineCount int32 `json:"degradedMachineCount"`
+
+	// Represents the latest available observations of current state.
+	Conditions []MachineConfigPoolCondition `json:"conditions"`
+}
+
+// MachineConfigPoolStatusConfiguration stores the current configuration for the pool, and
+// optionally also stores the list of machineconfig objects used to generate the configuration.
+type MachineConfigPoolStatusConfiguration struct {
+	corev1.ObjectReference `json:",inline"`
+
+	// source is the list of machineconfigs that were used to generate the single machineconfig object specified in `content`.
+	// +optional
+	Source []corev1.ObjectReference `json:"source,omitempty"`
+}
+
+// MachineConfigPoolCondition contains condition information for an MachineConfigPool.
+type MachineConfigPoolCondition struct {
+	// Type of the condition, currently ('Done', 'Updating', 'Failed').
+	Type MachineConfigPoolConditionType `json:"type"`
+
+	// Status of the condition, one of ('True', 'False', 'Unknown').
+	Status corev1.ConditionStatus `json:"status"`
+
+	// LastTransitionTime is the timestamp corresponding to the last status
+	// change of this condition.
+	LastTransitionTime metav1.Time `json:"lastTransitionTime"`
+
+	// Reason is a brief machine readable explanation for the condition's last
+	// transition.
+	Reason string `json:"reason"`
+
+	// Message is a human readable description of the details of the last
+	// transition, complementing reason.
+	Message string `json:"message"`
+}
+
+// MachineConfigPoolConditionType valid conditions of a machineconfigpool
+type MachineConfigPoolConditionType string
+
+const (
+	// MachineConfigPoolUpdated means machineconfigpool is updated completely.
+	// When the all the machines in the pool are updated to the correct machine config.
+	MachineConfigPoolUpdated MachineConfigPoolConditionType = "Updated"
```

Comment:
Not sure how much flexibility there is for renaming, but `Updated` doesn't seem quite right, since this is really "100% healthy", and could be recovering from a node's death or other reconcilable change and not just from a version change.  Maybe `Complete`?

Reply:
As you outlined Trevor, we aren't that free at changing the API 😄 we might end up changing this to complete and deprecate the old field (outside the scope of this PR fwiw)